server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -629,6 +638,17 @@ private ExtensionsSettings readFromExtensionsYml(Path filePath) throws IOExcepti
                         }
                     }
 
+                    ExtensionScopedSettings extAdditionalSettings = new ExtensionScopedSettings(additionalSettings);
+                    Set<String> additionalSettingsKeys = additionalSettings.stream().map(s -> s.getKey()).collect(Collectors.toSet());
```

Comment:
Minor, we probably don't need to compute `additionalSettingsKeys` for every extension, could be moved out from the loop, right?

Reply:
Done. CI just passed and now its running again. Hopefully this doesn't get affected by any flaky tests for this run.